tests/Core/Ruleset/SetSniffPropertyTest.php

```diff
@@ -366,6 +366,15 @@ public function testDirectCallWithOldArrayFormatThrowsDeprecationNotice()
             $exceptionClass = 'PHPUnit_Framework_Error_Deprecated';
         }
 
+        // @see https://github.com/sebastianbergmann/phpunit/issues/5062
+        set_error_handler(
+            static function ($errorNumber, $errorMessage, $file, $line) use ($exceptionClass) {
+                restore_error_handler();
+                throw new $exceptionClass($errorMessage, $errorNumber, $file, $line);
+            },
+            E_USER_DEPRECATED
+        );
```

Comment:
I've tested with various versions of PHPUnit; versions 9.5.9 and below work, but versions 9.5.10 and above do not. A quick search in the repository for this shows https://github.com/sebastianbergmann/phpunit/commit/fac02620f6b38ae54d47fe840e0095e68226a56c which has been included since version `9.5.10`. If you don't want to accept this change, then we'll need to limit the maximum version of PHPUnit to version 9.5.9 for now.

Reply:
I've been running locally for months on PHPUnit 9.5.10 and higher without problems, so there must be something else going on. More info needed.

Reply:
Thanks for following up. I've investigated further. Back when this project didn't work on PHP 8.x, I set up a script to run tests using the appropriate commands as extracted from the `.github/workflow/` files at the time. (Running `composer test` didn't work.) It looks like in https://github.com/PHPCSStandards/PHP_CodeSniffer/commit/10b01fe41017b7b1f88d24b5537ea2cab9bdd1c6 you made more changes than I noticed. I had noticed that it was safe for me to remove the `--ignore-platform-reqs` argument from my script when running `composer update`, but I hadn't noticed at the time that I should have updated the line that calls phpunit. When I run `vendor/bin/phpunit`, I get a warning reported (and the command returns success); however when I run the old command (`vendor/bin/phpunit tests/AllTests.php --no-configuration --bootstrap=tests/bootstrap.php --dont-report-useless-tests`), then test-suite fails and needs this change (or something similar) to make it pass. I've updated my test-runner script now.

Based on this (and your preference to not have this code here yet), I'll remove this change from this pull request.

Reply:
@fredden Ha, glad you figured it out! That old command was a work-around for running the tests on PHP 8.1+ with PHPUnit 7.x and should not be used anymore.